### PR TITLE
EVSS: Add a middleware to transform EVSS responses to snakecase

### PR DESCRIPTION
**Pull request by carlmjohnson**
Fixes https://github.com/department-of-veterans-affairs/sunsets-team/issues/236

Given how much this code touches, I don't think we should merge it before we do more testing.

### CODE REVIEW

File: app/serializers/disability_claim_base_serializer.rb [hunk 1]
```diff
@@ -17,31 +17,31 @@ def date_filed
   end
 
   def min_est_date
-    date_from_string 'minEstClaimDate'
+    date_from_string 'min_est_claim_date'
```
Comment (carlmjohnson) [inline app/serializers/disability_claim_base_serializer.rb]: One idea, create a thing like `dates_from_strings :min_est_claim_date, :max_est_claim_date` etc. that converts these en mass.

File: lib/evss/base_service.rb [hunk 2]
```diff
@@ -51,7 +52,7 @@ def conn
         faraday.use      :breakers
         faraday.use      EVSS::ErrorMiddleware
         faraday.use      Faraday::Response::RaiseError
-        faraday.response :json, content_type: /\bjson$/
+        faraday.use      EVSS::SnakeCaseMiddleware, content_type: /\bjson$/
```
Comment (saneshark) [inline lib/evss/base_service.rb]: Can this just use: 

`conn.response :snakecase`

from: https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/common/client/middleware/response/snakecase.rb

Comment (carlmjohnson) [inline lib/evss/base_service.rb]: Cool. Thanks, will use.

**Comment by carlmjohnson** [comment]
Closing while I redo it with @saneshark's thingie.

**Pull request closed without merging**